src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java

```diff
@@ -0,0 +1,446 @@
+import thaumcraft.common.Thaumcraft;
+import thaumcraft.common.lib.research.ResearchManager;
+import thaumcraft.common.tiles.TileInfusionMatrix;
+import thaumcraft.common.tiles.TilePedestal;
+
+public class TST_InfusionMaterialDispenser extends GTCM_MultiMachineBase<TST_InfusionMaterialDispenser> {
+
+    public TST_InfusionMaterialDispenser(int aID, String aName, String aNameRegional) {
+        super(aID, aName, aNameRegional);
+    }
+
+    public TST_InfusionMaterialDispenser(String aName) {
+        super(aName);
+    }
+
+    private TileInfusionMatrix targetMatrix;
+    private TilePedestal mainPedestal; //
+    private ArrayList<TilePedestal> subPedestals = new ArrayList<>();
+
+    public EntityPlayerMP fakePlayer = null;
+    public String playerName = null;// The names of the items inside the controller should be named after the real
+                                    // players' names.
+
+    public static final int STATE_IDLE = 0;
+    public static final int STATE_INFUSING = 1;
+    private boolean outputProcessed = false;
+    private IStructureDefinition<TST_InfusionMaterialDispenser> multiDefinition = null;
+    public int infusionState = STATE_IDLE;
+
+    @Override
+    public void saveNBTData(NBTTagCompound aNBT) {
+        aNBT.setInteger("infusionState", this.infusionState);
+        aNBT.setBoolean("outputProcessed", this.outputProcessed);
+        super.saveNBTData(aNBT);
+    }
+
+    @Override
+    public void loadNBTData(NBTTagCompound aNBT) {
+        this.infusionState = aNBT.getInteger("infusionState");
+        this.outputProcessed = aNBT.getBoolean("outputProcessed");
+        this.subPedestals.clear();
+        super.loadNBTData(aNBT);
+    }
+
+    // spotless:off
+    //In the logical section, code formatting is prohibited to avoid affecting readability.
+    @Override
+    public @NotNull CheckRecipeResult checkProcessing()  {
+        ArrayList<ItemStack> tItemsList = getStoredInputs();
+        //Set some basic parameters
+        setElectricityStats();
+
+        //Actually, it's not necessary to check every time, but who cares about this little expense?
+        World WORLD = this.getBaseMetaTileEntity().getWorld();
+        int x = this.getBaseMetaTileEntity().getXCoord();
+        int y = this.getBaseMetaTileEntity().getYCoord();
+        int z = this.getBaseMetaTileEntity().getZCoord();
+
+        TileEntity tempTile = null;
+         tempTile = WORLD.getTileEntity(x, y-1, z );
+         //Check if there is an infusion matrix beneath the main unit.
+         if (!(tempTile instanceof TileInfusionMatrix)) {
+             return SimpleCheckRecipeResult.ofFailure("Can't find infusion matrix");
+         }
+         targetMatrix = (TileInfusionMatrix) tempTile;
+
+         //Check if there is an Unactivated infusion matrix .
+        if(!targetMatrix.active){
+            return SimpleCheckRecipeResult.ofFailure("Unactivated infusion matrix");
```

Comment:
same to the above.
and there are more below, please make sure to fix all of them.